src/serializer.js

```diff
@@ -97,38 +96,39 @@ var Serializer = function () {
   __Serializer.prototype.makeUpPrefix = function (uri) {
     var p = uri
     var pok
-    function canUse (pp) {
-      if (! __Serializer.prototype.validPrefix.test(pp)) return false // bad format
+    function canUseMethod (pp) {
+      if (!__Serializer.prototype.validPrefix.test(pp)) return false // bad format
       if (pp === 'ns') return false // boring
       if (pp in this.namespaces) return false // already used
       this.prefixes[uri] = pp
       this.namespaces[pp] = uri
       pok = pp
       return true
     }
-    canUse = canUse.bind(this)
-    /*    for (var ns in this.prefixes) {
-            namespaces[this.prefixes[ns]] = ns // reverse index foo
-        }
-        */
+    var canUse = canUseMethod.bind(this)
+
     if ('#/'.indexOf(p[p.length - 1]) >= 0) p = p.slice(0, -1)
     var slash = p.lastIndexOf('/')
     if (slash >= 0) p = p.slice(slash + 1)
     var i = 0
-    while (i < p.length)
-    if (this.prefixchars.indexOf(p[i])) i++; else
+    while (i < p.length) {
+      if (this.prefixchars.indexOf(p[i])) {
+        i++
+      } else {
         break
+      }
+    }
     p = p.slice(0, i)
     if (p.length < 6 && canUse(p)) return pok // exact i sbest
     if (canUse(p.slice(0, 3))) return pok
     if (canUse(p.slice(0, 2))) return pok
     if (canUse(p.slice(0, 4))) return pok
     if (canUse(p.slice(0, 1))) return pok
     if (canUse(p.slice(0, 5))) return pok
-    if (! __Serializer.prototype.validPrefix.test(p)) {
+    if (!__Serializer.prototype.validPrefix.test(p)) {
       p = 'n' // Otherwise the loop below may never termimnate
     }
-    for (var i = 0;; i++) if (canUse(p.slice(0, 3) + i)) return pok
+    for (var j = 0; ; j++) if (canUse(p.slice(0, 3) + j)) return pok
```

Comment:
Good point. Yes, I re-coded it, as  (canUse(p.slice(0, 3) + i)) return p.slice(0, 3) + i)   etc